Source/JavaScriptCore/b3/B3LowerToAir.cpp

```diff
@@ -3796,8 +3816,16 @@ class LowerToAir {
             default:
                 RELEASE_ASSERT_NOT_REACHED();
             }
-            append(op, scalar, tmp(value));
-            return;
+            if (isValidForm(op, Arg::Tmp, Arg::Tmp)) {
+                append(op, scalar, tmp(value));
+                return;
+            }
+            if (isValidForm(op, Arg::Tmp, Arg::Tmp, Arg::Tmp)) {
+                RELEASE_ASSERT(isX86());
```

Comment:
Is this needed?

Reply:
Huh. Doesn't seem like we do, I'll make sure there wasn't some failure we ran into that provoked this though.

Reply:
Removed it and everything still works, as one would expect in hindsight. Good catch!